### DISCUSSION TITLE
"Insufficient RU balance" error

### DISCUSSION THREAD

**Discussion by luccamordente** [WalletConnect/walletconnect-monorepo#3849] [Developer Support]
I'm getting this error when connecting a wallet and can't find any clue about it.
Happens intermittently. Any ideas?

```
Uncaught (in promise) InvalidInputRpcError: Missing or invalid parameters.
Double check you have provided the correct parameters.

URL: https://rpc.walletconnect.com/v1/?chainId=eip155:11155111&projectId=182619aee5f98b0b86eccd8b6c2302fa
Request body: {"method":"eth_getBalance","params":["0x1D9fdA9873AA00B9F58C5D7C261760E5855e2415","latest"]}

Details: Insufficient RU balance
Version: viem@1.16.5
```

Not sure if it's related, but been also intermittently (but very frequently) getting zero balance for SepoliaETH and other tokens.

**Comment by SergeyVolynkin** [comment]
cc @glitch-txs @0xAsimetriq when you'll have time  🙏

**Comment by edmundedgar** [comment]
This is likely coming from the RPC endpoint. BlockPi calls its credits "Request Units" or "RU". If you hit it too hard it'll decline to answer your RPC requests, and anything you try to query from the block chain will come back as either zero or an error.
